### PR TITLE
Add platform label to balo name

### PR DESCRIPTION
**Pull request by pramodya1994**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/27285

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java [hunk 1]
```diff
@@ -258,6 +258,39 @@ public void testBuildBalProject() throws IOException {
                 .resolve("winery.bir").toFile().exists());
     }
 
+    @Test(description = "Build a valid ballerina project with platform dependencies")
+    public void testBuildBalProjectWithPlatformDependencies() throws IOException {
```
Comment (hevayo) [inline cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java:262]: Shall we add a test without platform dependancies

Comment (hevayo) [inline cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java:262]: Shall we add a test without platform dependencies

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java [hunk 2]
```diff
@@ -181,11 +182,10 @@ public static String guessPkgName(String packageName) {
     }
 
     public static String getBaloName(PackageManifest pkgDesc) {
-        return ProjectUtils.getBaloName(pkgDesc.org().toString(),
-                pkgDesc.name().toString(),
-                pkgDesc.version().toString(),
-                null
-        );
+        return ProjectUtils.getBaloName(pkgDesc.platform(JdkVersion.JAVA_11.code()),
```
Comment (hevayo) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java:185]: Shouldn't the platform be the last param according to getBaloName in 191 like

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java [hunk 3]
```diff
@@ -196,6 +196,17 @@ public static String getBaloName(String org, String pkgName, String version, Str
         return org + "-" + pkgName + "-" + platform + "-" + version + BLANG_COMPILED_PKG_BINARY_EXT;
     }
 
+    public static String getBaloName(PackageManifest.Platform platform, String org, String pkgName, String version) {
```
Comment (hevayo) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java:199]: Isnt this redudent with 191 function

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/PackageResolutionTests.java [hunk 4]
```diff
@@ -143,7 +143,7 @@ public void testProjectWithMissingTransitiveDependency() throws IOException {
         // package_missing_transitive_dep --> package_k --> package_z (this is missing)
         Path baloPath = RESOURCE_DIRECTORY.resolve("balos").resolve("missing_transitive_deps")
                 .resolve("samjs-package_k-any-1.0.0.balo");
-        BCompileUtil.copyBaloToDistRepository(baloPath, "samjs", "package_k", "1.0.0");
+        BCompileUtil.copyBaloToDistRepository(baloPath, "samjs", "package_k", "1.0.0", null);
```
Comment (hevayo) [inline project-api/project-api-test/src/test/java/io/ballerina/projects/test/PackageResolutionTests.java:146]: Why do we have a extra null

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBaloWriter.java [hunk 5]
```diff
@@ -270,9 +270,11 @@ public void testBaloWriterWithTwoDirectDependencies() throws IOException {
 
         PackageCompilation packageCompilation = project.currentPackage().getCompilation();
         Target target = new Target(project.sourceRoot());
-        String baloName = ProjectUtils.getBaloName(project.currentPackage().packageOrg().toString(),
-                project.currentPackage().packageName().toString(), project.currentPackage().packageVersion().toString(),
-                null);
+        String baloName = ProjectUtils.getBaloName(
```
Comment (hevayo) [inline project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBaloWriter.java:273]: Shall we have a method to get balo name from package.

**Review by hevayo — CHANGES_REQUESTED** [review]

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java [hunk 6]
```diff
@@ -258,6 +258,39 @@ public void testBuildBalProject() throws IOException {
                 .resolve("winery.bir").toFile().exists());
     }
 
+    @Test(description = "Build a valid ballerina project with platform dependencies")
```
Comment (praneesha) [inline cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java:261]:
```suggestion
    @Test(description = "Build a valid Ballerina project with platform dependencies")
```

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java [hunk 7]
```diff
@@ -258,6 +258,39 @@ public void testBuildBalProject() throws IOException {
...
+    public void testBuildBalProjectWithPlatformDependencies() throws IOException {
+        Path projectPath = this.testResources.resolve("validProjectWithPlatformDependencies");
+        System.setProperty("user.dir", projectPath.toString());
+        BuildCommand buildCommand = new BuildCommand(projectPath, printStream, printStream, false, true);
+        // non existing bal file
```
Comment (praneesha) [inline cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java:266]:
```suggestion
        // Non-existing BAL file.
```

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBaloWriter.java [hunk 8]
```diff
@@ -270,9 +270,11 @@ public void testBaloWriterWithTwoDirectDependencies() throws IOException {
...
+                project.currentPackage().manifest().platform(JdkVersion.JAVA_11.code()),
+                project.currentPackage().packageOrg().toString(),
+                project.currentPackage().packageName().toString(),
+                project.currentPackage().packageVersion().toString());
         Path baloPath = target.getBaloPath().resolve(baloName);
         // balo name
```
Comment (praneesha) [inline project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBaloWriter.java:279]:
```suggestion
        // BALO name
```

File: tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BCompileUtil.java [hunk 9]
```diff
@@ -151,15 +155,16 @@ private static boolean isSingleFileProject(Project project) {
...
             Path baloDirPath = distributionCache.resolve("balo")
                     .resolve(org)
                     .resolve(pkgName)
                     .resolve(version);
             Files.createDirectories(baloDirPath);
-            String baloFileName = ProjectUtils.getBaloName(org, pkgName, version, null);
+            String baloFileName = ProjectUtils.getBaloName(platform, org, pkgName, version);
             return baloDirPath.resolve(baloFileName);
         } catch (IOException e) {
             throw new RuntimeException("error while creating the balo distribution cache directory at " +
```
Comment (praneesha) [inline tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BCompileUtil.java:170]:
```suggestion
            throw new RuntimeException("Error while creating the BALO distribution cache directory at " +
```

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java [hunk 10]
```diff
@@ -258,6 +258,39 @@ public void testBuildBalProject() throws IOException {
...
+        Assert.assertTrue(projectPath.resolve("target").resolve("bin").resolve("winery.jar").toFile().exists());
+        Assert.assertTrue(projectPath.resolve("target").resolve("cache").resolve("foo")
+                                  .resolve("winery").resolve("0.1.0").resolve("java11")
+                                  .resolve("winery.jar").toFile().exists());
+        Assert.assertTrue(projectPath.resolve("target").resolve("cache").resolve("foo")
+                                  .resolve("winery").resolve("0.1.0").resolve("bir")
+                                  .resolve("winery.bir").toFile().exists());
+    }
+
     @Test(description = "Build a valid ballerina project")
```
Comment (praneesha) [inline cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/BuildCommandTest.java:294]:
```suggestion
    @Test(description = "Build a valid Ballerina project")
```

**Pull request closed without merging**